guides/micronaut-security-x509/micronaut-security-x509.adoc

```diff
@@ -0,0 +1,444 @@
+common:header-top.adoc[]
+
+== Getting Started
+
+In this guide, we will create a Micronaut application with X.509 authentication written in @language@.
+
+== What you will need
+
+To complete this guide, you will need the following:
+
+* Some time on your hands
+* A decent text editor or IDE
+* JDK 1.8 or greater installed with `JAVA_HOME` configured appropriately
+* https://www.openssl.org/[OpenSSL] installed
+
+common:completesolution.adoc[]
+
+== X.509 Mutual Authentication
+
+As of Micronaut Security version 3.3, applications can use https://en.wikipedia.org/wiki/X.509[X.509] client certificates when using SSL to enable https://en.wikipedia.org/wiki/Mutual_authentication[mutual authentication], i.e. in addition to the client verifying the server identity with its SSL certificate, the server can verify the client identity with a client certificate.
+
+== Create certificates for testing
+
+First, we'll need a Certificate Authority (CA), an SSL certificate for the server, and an X.509 certificate for each application user who will be using mutual authentication. You can use an existing CA, server certificate, and/or client certificate if you already have them, otherwise create your own as described below.
```

Comment:
```suggestion
First, we'll need a Certificate Authority (CA), an SSL certificate for the server, and an X.509 certificate for each application user who will be using mutual authentication. You can use an existing CA, server certificate, and/or client certificate if you already have them; otherwise create your own as described below.
```